src/Neo.CLI/CLI/MainService.cs

```diff
@@ -390,13 +390,18 @@ public async void Start(string[] args)
 
             using (IEnumerator<Block> blocksBeingImported = GetBlocksFromFile().GetEnumerator())
             {
+                ThreadPool.GetMaxThreads(out var maxWorkerThreads, out _);
                 while (true)
                 {
                     List<Block> blocksToImport = new List<Block>();
-                    for (int i = 0; i < 10; i++)
+                    for (int i = 0; i < maxWorkerThreads;)
                     {
                         if (!blocksBeingImported.MoveNext()) break;
-                        blocksToImport.Add(blocksBeingImported.Current);
+                        var block = blocksBeingImported.Current;
+                        blocksToImport.Add(block);
+                        if (block.Transactions.Length == 0) continue;
```

Comment:
if we have 100000 blocks without tx we will send all together?

Reply:
Two times of the maximum system threads at most if it contains empty blocks. If all contains txs, then maximum system thread.

Reply:
We can chunk them by 1000 or similar, imagine 1 millions blocks without transactions, it could produce an out of memory